app/models/manageiq/providers/lenovo/manager_mixin.rb

```diff
@@ -73,7 +83,15 @@ def discover_queue(username, password, zone = nil)
     private
 
     def discover_from_queue(username, password)
-      discover(username, MiqPassword.decrypt(password))
+      discover(username, '')
```

Comment:
So, `discover_queue` puts [`discover_from_queue` onto MiqQueue](https://github.com/ManageIQ/manageiq-providers-lenovo/pull/61/files#diff-9fd48102305199276ae849195d9a47d8R77).

This will be called later, probably by a GenericWorker.

Currently, `discovere_queue` is receiving `username` and `password`, but I suspect it should also receive `ip/hostname` and `port`.  In fact, it should *at least* receive `ip/hostname` and `port`.

Step back and think about the end user.  They'll be presented with a page asking to discover an LXCA server.  If they just enter `username` and `password`, the code would have to scan the entire network looking for LXCA servers, and attempting the username and password on each.

Instead, if the user is allowed to enter an `ip` and `port`, the code just has to attempt to connect to that ip and port and see if it's an LXCA server.  If the user also supplies a `username` and `password`, the code can attempt to authenticate as well.

Another idea (something that's done for VMWare, I believe) is to allow the user to pass in an Subnet Mask.  Then the code can search a range of IPs looking for LXCA servers.  If you decide to do that, I think the `username` and `password` would not be needed.  Since it's unlikely that if the code discovers two LXCA servers in that IP range, that they will both use the same credentials.